### PR TITLE
docs: Add conda-forge badge to README

### PR DESCRIPTION
**Pull request by matthewfeickert**
* c.f. https://github.com/conda-forge/func-adl-feedstock

[conda-forge version](https://github.com/conda-forge/func-adl-feedstock.git)

### CODE REVIEW

**Comment by matthewfeickert** [comment]
@gordonwatts this is ready for review.

**Comment by matthewfeickert** [comment]
@gordonwatts ping on this.

**Comment by matthewfeickert** [comment]
@gordonwatts ping on this.

**Comment by matthewfeickert** [comment]
@gordonwatts ping on this.

**Comment by gordonwatts** [comment]
Sorry - I can add this. But note that I know nothing about conda forge (and I apologize for not yet seeing the motivation for this). :-)

**Comment by matthewfeickert** [comment]
> But note that I know nothing about conda forge (and I apologize for not yet seeing the motivation for this). :-)

@gordonwatts Thanks for the merge. I think that's totally okay to not be well versed in all the different distribution methods of Python libraries. It isn't something that is _yet_ super relevant in our ATLAS world given the very tight coupling of an entire C++ computing environment to particular Python binaries, rather than targeting ABIs which would provide greater flexibility. (There's motivations for this that may or may not actually be relevant any more that you know better than I but is also not relevant for this Issue).

The motivation for the badge is to communicate to people that `func_adl` exists on conda-forge and that the version on conda-forge is kept in-sync with the PyPI version.

The motivation for having distributions be on conda-forge in the first place is that conda-forge uses a bootstrap system where all dependencies of a package must first exist on conda-forge to have a clear build provenance. This means that if simple but foundational libraries are missing then you can't build more complex / user facing packages.

The motivation for getting things from conda-forge as an alternative to PyPI alone is that you can specify all dependencies of your software environment down to the compiler level on conda-forge, which when using a tool like `pixi` allows you to trivially get fully reproducible environments at the hash level from the automatically produced lock file and be able to distribute them. Additionally, this also allows for more coherent environment building when creating software distributions in Linux containers (for example). Instead of having to mix different dependency solver solutions (`conda`/`libmamba` and `pip`/`uv`) and hope that they agree (which they very reasonably might not) you can solve the entire software stack that you're deploying from a single environment solved by a single solver. In addition to being more reproducible (if you're not using a lock file) it also alleviates large sources of environment debugging pain that might not be reproducible across platforms.

TL/DR:

Distributing across PyPI and conda-forge allows for people to make better choices based off the environment and tooling they have and leads to more reproducible environments.
